test_tf.py

```diff
@@ -0,0 +1,153 @@
+import os
+import subprocess
+import unittest
+from main import get_policy_check_type, get_required_inputs, get_optional_inputs, get_sub_command
+from main import get_treat_findings_as_non_blocking_flag, build_command, execute_command, set_output
+
+CLI_POLICY_VALIDATOR = "tf-policy-validator"
+
+POLICY_CHECK_TYPE = "INPUT_POLICY-CHECK-TYPE"
+
+NOT_VALIDATE_POLICY = "NOT_VALIDATE_POLICY"
+VALIDATE_POLICY = "VALIDATE_POLICY"
+CHECK_NO_NEW_ACCESS = "CHECK_NO_NEW_ACCESS"
+CHECK_ACCESS_NOT_GRANTED = "CHECK_ACCESS_NOT_GRANTED"
+
+COMMON_REQUIRED_INPUTS = {"INPUT_TEMPLATE-PATH", "INPUT_REGION"}
+
+TREAT_FINDINGS_AS_NON_BLOCKING = "INPUT_TREAT-FINDINGS-AS-NON-BLOCKING"
+
+# assertRaises(SomeCoolException, mymod.myfunc)
+class TfpvTest(unittest.TestCase):
+    # test_get_type_case1: failure expected because required os.environ[] are not set")
+    @unittest.expectedFailure
+    def test_get_type_case1(self):
+        assertRaises(ValueError, get_policy_check_type)
```

Comment:
is `@unittest.expectedFailure` supposed to be used together with `assertRaises` ?

Reply:
yes else test fails.

Reply:
I think this should assert that `KeyError` is raised if the environment variable is not set (raised [here](https://github.com/hlijuen/terraform-aws-iam-policy-validator/blob/8212de2eef5935ac8cde9f4e9fa6578fb7beeb81/main.py#L77) when `os.environ[POLICY_CHECK_TYPE]` is not set), instead of expecting the test to fail

Reply:
yeah, I agree with Alan.

Reply:
@alankuo-aws
The source code (main.py, at lline 80) raises ValueError, thus it should be ValueError unless we change main.py.
@mponaws
action.yaml lists a bunch of input parameters. There is no environment variable for instance POLICY_CHECK_TYPE specified in that file.

Reply:
I think we'd want to validate that expected exception is raised, rather than expecting the test to fail; so using assertRaises together with unittest.expectedFailure is an anti pattern.

regarding the usage of os.environ[], we should use `os.getenv` - reference: https://stackoverflow.com/questions/65854207/keyerror-with-os-environ-accessing-variable-from-env-file. (not a goal for this CR, we can refactor later)